src/rpc_client/src/meta_client.rs

```diff
@@ -1221,17 +1224,57 @@ impl GrpcMetaClient {
             .map_err(RpcError::TransportError)
     }
 
-    pub(crate) async fn try_build_rpc_channel(addrs: Vec<String>) -> Result<(Channel, String)> {
+    fn get_retry_strategy() -> impl Iterator<Item = Duration> {
+        ExponentialBackoff::from_millis(Self::CONN_RETRY_BASE_INTERVAL_MS)
+            .max_delay(Duration::from_millis(Self::CONN_RETRY_MAX_INTERVAL_MS))
+            .map(jitter)
+    }
+
+    /// Connect against one of multiple meta nodes using a K8s service
+    pub(crate) async fn try_build_rpc_channel_service(addr: String) -> Result<(Channel, String)> {
+        let endpoint = Self::addr_to_endpoint(addr.clone())?;
+
+        let res = tokio_retry::Retry::spawn(Self::get_retry_strategy(), || async {
+            let endpoint = endpoint.clone();
+            let addr = addr.clone();
+            match Self::connect_to_endpoint(endpoint).await {
+                Ok(channel) => {
+                    tracing::info!(
+                        "Connect to meta server via service at {} successfully",
+                        addr
+                    );
+                    Ok((channel, addr))
+                }
+                Err(e) => {
+                    tracing::warn!(
+                        "Failed to connect to meta server using service at {}, trying again: {}",
```

Comment:
🤔  It looks exactly the same as `try_build_rpc_channel_servers` except this warning log. Do we really need to introduce a new function here in order to print different logs?

Reply:
Or we can simply use logs like `Failed to connect to meta server via {}, trying again: {}` for these two strategy to avoid misleading.

Reply:
For me the problem is the return code

```
Err(RpcError::Internal(anyhow!(
    "Failed to connect to any meta server"
)))
```

If we are using `try_build_rpc_channel` with a service and we happen to connect against the downed meta node, then we will return above error. `GrpcMetaClient::new` will then fail, even though there may be alive meta nodes out there.